### PR TITLE
fix(legend): width with scroll bar

### LINKED ISSUE #1011: Legend sizing does not account for scrollbar width
**Describe the bug**
The `Right` or `Left` legend positions do not account for the browser scrollbar. Particularly, with short legend names.

Only on Chrome and Safari, not FireFox.

**To Reproduce**
1. Go to https://elastic.github.io/elastic-charts/?path=/story/small-multiples-alpha--grid-lines&knob-Show Legend=true
2. Toggle the legend with the provided knob
3. Notice the legend item labels are truncated due to the scroll bar.

**Expected behavior**
Somehow account for the scrollbar width when legend values exceed the height of the legend container.

**Screenshots**
Screen Recording 2021-02-03 at 09 35 AM

When the legend container is set to `overflow-x: hidden` the issue goes away

image

**Kibana Cross Issues**
https://github.com/elastic/kibana/issues/94596

### PR DESCRIPTION
**Pull request by nickofthyme**
## Summary

fixes #1011

Checks to see if total legend height on vertical legends, is greater than parent container height. If so adds scrollbar width to total legend width.

### Before
image

### After
image

### Checklist
- [x] This was checked for cross-browser compatibility
- [x] Unit tests were updated or added to match the most common scenarios

### CODE REVIEW

**Review by markov00 — APPROVED** [review]
Looks great, I've just nitpicked about the legend size calculation

File: src/components/legend/_legend.scss [hunk 1]
```diff
@@ -34,7 +28,7 @@
   }
 
   &--debug {
-    background: red;
+    background: rgba(238, 130, 238, 0.2);
```
[the remark this answers is not in the document]
Comment (nickofthyme) [reply src/components/legend/_legend.scss:31]: I was sick of seeing it 🤣 . I would scare me every time I saw it 😨

File: src/state/selectors/get_legend_size.ts [hunk 2]
```diff
@@ -83,8 +84,12 @@ export const getLegendSizeSelector = createCachedSelector(
       MARKER_WIDTH + MARKER_LEFT_MARGIN + bbox.width + (showLegendDisplayValue ? VALUE_LEFT_MARGIN : 0);
     if (isVerticalAxis(position)) {
       const legendItemHeight = bbox.height + VERTICAL_PADDING * 2;
+      const scollBarDimension = legendItemHeight * labels.length > parentDimensions.height ? SCROLL_BAR_WIDTH : 0;
```
[the remark this answers is not in the document]
Comment (nickofthyme) [reply src/state/selectors/get_legend_size.ts]: Yeah you are correct about the top margin. The padding on the `<li>` is now `2px` but there is a new `.background` element that has a top and bottom margin of `2px` to fix the cursor highlighting issue. So I think for simplicity we keep the `VERTICAL_PADDING` combined. See ea8c853.

image

**Comment by nickofthyme** [comment]
jenkins retest this please

**Pull request merged**